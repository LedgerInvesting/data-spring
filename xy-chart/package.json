{
  "name": "@data-spring/xy-chart",
  "version": "0.9.6",
  "main": "build/index.js",
  "scripts": {
    "build": "webpack"
  },
  "description": "Components and utility functions for using react-spring animations in data-ui/xy-chart data series visualizations https://ledgerinvesting.github.io/data-spring/",
  "repository": "https://github.com/LedgerInvesting/data-spring.git",
  "keywords": [
    "d3",
    "react",
    "chart",
    "visualization",
    "plot",
    "motion",
    "animated",
    "animation",
    "spring",
    "data-ui",
    "react-spring"
  ],
  "author": "Matt Liu <matt@ledgerinvesting.com>",
  "license": "MIT",
  "dependencies": {
    "@babel/runtime": "^7.0.0"
  },
  "devDependencies": {
    "babel-core": "^6.26.3",
    "babel-loader": "^7.1.5",
    "babel-eslint": "^8.2.2",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "@data-ui/xy-chart": "0.0.54",
    "react-spring": "^7.2.8",
    "react": "^16.0.0",
    "react-dom": "^16.0.0",
    "eslint": "^4.19.1",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-react": "^7.7.0",
    "webpack": "^4.20.2",
    "webpack-cli": "^3.1.1"
  }
}
