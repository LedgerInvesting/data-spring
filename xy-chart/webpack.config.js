var path = require('path');

module.exports = {
    mode: 'production',
    entry: './src/index.js',
    output: {
        path: path.resolve('build'),
        filename: 'index.js'
    },
    module: {
        rules: [
            {
                test: /\.jsx?$/,
                exclude: /(node_modules)/,
                use: 'babel-loader'
            }
        ]
    }
}
