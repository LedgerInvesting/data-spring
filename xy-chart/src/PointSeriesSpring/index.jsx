import React from 'react';
import { PointSeries } from '@data-ui/xy-chart';
import { Spring, config } from 'react-spring';
import { calcFlattenedStart, constructArray } from '../../utils/motionUtil';

const PointSeriesSpring = (props) => {
  const motionPoints = calcFlattenedStart(props);

  return (
    <Spring
      {...props}
      config={props.springConfig}
      from={motionPoints.from}
      to={motionPoints.to}
    >
      {values => {
        return (
          <PointSeries
            {...props}
            size={values.size}
            data={constructArray(values, props.data)}
          />
        );
      }}
    </Spring>
  );
};

PointSeriesSpring.displayName = 'PointSeriesSpring';
PointSeriesSpring.defaultProps = {
  animationStart: 'bottomLeft',
  sizeCalculator: (point, props) => props.size,
  springConfig: config.default,
  customAnimationStart: (point) => point,
};
export default PointSeriesSpring;
