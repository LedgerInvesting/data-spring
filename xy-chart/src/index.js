import PointSeriesSpring from './PointSeriesSpring/index.jsx';
import LineSeriesSpring from './LineSeriesSpring/index.jsx';
import AreaSeriesSpring from './AreaSeriesSpring/index.jsx';

export {
  PointSeriesSpring as PointSeriesSpring,
  LineSeriesSpring as LineSeriesSpring,
  AreaSeriesSpring as AreaSeriesSpring,
};
