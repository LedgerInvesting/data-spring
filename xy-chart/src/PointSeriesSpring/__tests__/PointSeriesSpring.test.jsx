import React from 'react';
import PointSeriesSpring from '../';
import { config } from 'react-spring';
import { XYChart } from '@data-ui/xy-chart';
import renderer from 'react-test-renderer';


it('renders successfully', () => {
  const tree = renderer
    .create(
      <XYChart
        width={700}
        height={350}
        xScale={{ type: 'linear', domain: [0, chartBounds.xMax] }}
        yScale={{ type: 'linear', domain: [0, chartBounds.yMax] }}
      >
        <PointSeriesSpring
          strokeWidth={0.3}
          fillOpacity=".75"
          fill="#000"
          stroke="#000"
          animationStart="bottom"
          data={testSeries.xySeries}
          bounds={chartBounds}
          springConfig={config.gentle}
        />
      </XYChart>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with default animationStart props', () => {
  const tree = renderer
    .create(
      <XYChart
        width={700}
        height={350}
        xScale={{ type: 'linear', domain: [0, chartBounds.xMax] }}
        yScale={{ type: 'linear', domain: [0, chartBounds.yMax] }}
      >
        <PointSeriesSpring
          strokeWidth={0.3}
          fillOpacity=".75"
          fill="#000"
          stroke="#000"
          data={testSeries.xySeries}
          bounds={chartBounds}
          springConfig={config.gentle}
        />
      </XYChart>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with default springConfig props', () => {
  const tree = renderer
    .create(
      <XYChart
        width={700}
        height={350}
        xScale={{ type: 'linear', domain: [0, chartBounds.xMax] }}
        yScale={{ type: 'linear', domain: [0, chartBounds.yMax] }}
      >
        <PointSeriesSpring
          strokeWidth={0.3}
          fillOpacity=".75"
          fill="#000"
          stroke="#000"
          animationStart="bottom"
          data={testSeries.xySeries}
          bounds={chartBounds}
        />
      </XYChart>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const testSeries = {
  xySeries: [
    {
      x: 5,
      y: 6,
    },
    {
      x: 8,
      y: 3,
    },
    {
      x: 2,
      y: 7,
    },
    {
      x: 10,
      y: 1,
    },
  ],
}

const chartBounds = {
  xMin: 0,
  yMin: 0,
  xMax: 9,
  yMax: 10,
};
