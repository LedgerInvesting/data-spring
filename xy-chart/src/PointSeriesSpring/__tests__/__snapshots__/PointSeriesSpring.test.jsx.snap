// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<svg
  height={350}
  role="img"
  width={700}
>
  <g
    className="cx-group"
    transform="translate(64, 64)"
  >
    <g
      className="cx-group"
      style={null}
      transform="translate(0, 0)"
    >
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={317.77777777777777}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={508.4444444444444}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={127.1111111111111}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={635.5555555555555}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
    </g>
  </g>
</svg>
`;

exports[`renders with default animationStart props 1`] = `
<svg
  height={350}
  role="img"
  width={700}
>
  <g
    className="cx-group"
    transform="translate(64, 64)"
  >
    <g
      className="cx-group"
      style={null}
      transform="translate(0, 0)"
    >
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={0}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={0}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={0}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={0}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
    </g>
  </g>
</svg>
`;

exports[`renders with default springConfig props 1`] = `
<svg
  height={350}
  role="img"
  width={700}
>
  <g
    className="cx-group"
    transform="translate(64, 64)"
  >
    <g
      className="cx-group"
      style={null}
      transform="translate(0, 0)"
    >
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={317.77777777777777}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={508.4444444444444}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={127.1111111111111}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
      <a
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        role="presentation"
        xlinkHref="#"
      >
        <g
          className="cx-group vx-glyph"
          transform="translate(0, 0)"
        >
          <circle
            className="vx-glyph-dot"
            cx={635.5555555555555}
            cy={222}
            fill="#000"
            fillOpacity=".75"
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            r={4}
            stroke="#000"
            strokeDasharray={null}
            strokeWidth={0.3}
          />
        </g>
      </a>
    </g>
  </g>
</svg>
`;
